drivers/video/video_stm32_venc.c

```diff
@@ -0,0 +1,1026 @@
+/*
+ * Copyright (c) 2025 Hugues Fruchet <hugues.fruchet@foss.st.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_venc
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/irq.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/video.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/clock_control.h>
+
+#include "ewl.h"
+#include "h264encapi.h"
+#include "stm32n6xx_ll_venc.h"
+#include "reg_offset_v7.h"
+
+#include <zephyr/mem_mgmt/mem_attr_heap.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr-sw.h>
+
+LOG_MODULE_REGISTER(video_stm32_venc, CONFIG_VIDEO_LOG_LEVEL);
+
+#define EWL_HEAP_ALIGNED_ALLOC(size)				\
+	mem_attr_heap_aligned_alloc((ATTR_SW_ALLOC_NON_CACHE <<	\
+				    DT_MEM_SW_ATTR_SHIFT), ALIGNMENT_INCR, size)
+#define EWL_HEAP_ALIGNED_FREE(block) mem_attr_heap_free(block)
+
+#define VENC_RATECTRL_FPS 30
+#define VENC_RATECTRL_QP 25
+
+#define TRACE_EWL
+
+#define EWL_USE_POLLING_SYNC
+
+#define EWL_TIMEOUT 100UL
+
+#define ALIGNMENT_INCR 8UL
+
+#define MEM_CHUNKS      32
+
+#ifdef TRACE_EWL
+u32 mem_counter = 0;
+#endif /* TRACE_EWL */
+
+typedef struct
+{
+  u32 clientType;
+  /* tables required for buffer alignment */
+  u32 *chunks[MEM_CHUNKS];
+  u32 *alignedChunks[MEM_CHUNKS];
+  u32 totalChunks;
+
+  struct k_sem complete;
+  i32 status;
+  i32 irq_status;
+  i32 irq_cnt;
+} VENC_EWL_TypeDef;
+
+#define ALIGNED_SIZE(x) ((((x) + ALIGNMENT_INCR - 1UL) / ALIGNMENT_INCR) * ALIGNMENT_INCR)
```

Comment:
Use the Zephyr implementation (from `#include <zephyr/sys/util.h>`)
```suggestion
#define ALIGNED_SIZE(x) ROUND_UP(x, ALIGNMENT_INCR)
```
(or use a raw `ROUND_UP` everywhere in place of `ALIGNED_SIZE` - as you prefer)